contracts/oracles/Oracle.sol

```diff
@@ -0,0 +1,204 @@
+     */
+    function _addPaths(address[][] calldata paths_) internal {
+        for (uint256 i = 0; i < paths_.length; i++) {
+            require(paths_[i].length >= 2, "Oracle: path must be longer than 2");
+
+            for (uint256 j = 0; j < paths_[i].length - 1; j++) {
+                (bool isExist, address pair) = _isPairExistAtUniswap(
+                    paths_[i][j],
+                    paths_[i][j + 1]
+                );
+                require(isExist, "Oracle: uniswap pair doesn't exist");
+                _pairs.add(pair);
+                _incrementCounter(pair);
+            }
+            _paths[paths_[i][0]] = paths_[i];
+        }
+        updatePrices();
+    }
+
+    /**
+     * @dev Removes multiple token paths from the oracle.
+     * @param tokenIns_ The array of token addresses to remove.
+     */
+    function _removePaths(address[] calldata tokenIns_) internal {
+        for (uint256 i = 0; i < tokenIns_.length; i++) {
+            for (uint256 j = 0; j < _paths[tokenIns_[i]].length - 1; j++) {
+                address pair = uniswapV2Factory.getPair(
+                    _paths[tokenIns_[i]][j],
+                    _paths[tokenIns_[i]][j + 1]
+                );
+
+                if (_pairCounters[pair] == 1) _pairs.remove(pair);
+                _decrementCounter(pair);
+            }
+            delete _paths[tokenIns_[i]];
+        }
+    }
+
+    function _incrementCounter(address pair_) internal {
+        _pairCounters[pair_] = _pairCounters[pair_] + 1;
+    }
+
+    function _decrementCounter(address pair_) internal {
+        if (_pairCounters[pair_] > 0) _pairCounters[pair_] = _pairCounters[pair_] - 1;
+    }
+
+    function _isPairExistAtUniswap(
+        address token1_,
+        address token2_
+    ) internal view returns (bool, address) {
+        address pair_ = uniswapV2Factory.getPair(token1_, token2_);
+        if (pair_ == address(0)) return (false, pair_);
+        return (true, pair_);
+    }
+
+    function _getPrice(address pair_, address expectedToken_) internal view returns (uint256) {
+        PairInfo storage pairInfo = _pairInfos[pair_];
+
+        if (pairInfo.blockTimestamps.length == 0) {
+            return 0;
+        }
+
+        uint256 index_ = pairInfo.blockTimestamps.lowerBound(block.timestamp - timeWindow);
+        index_ = index_ == 0 ? index_ : index_ - 1;
+
+        uint256 price0CumulativeOld = pairInfo.prices0Cumulative[index_];
+        uint256 price1CumulativeOld = pairInfo.prices1Cumulative[index_];
+        uint256 blockTimestampOld = pairInfo.blockTimestamps[index_];
+
+        uint256 price0_;
+        uint256 price1_;
+
+        unchecked {
+            (uint256 price0Cumulative, uint256 price1Cumulative, uint256 blockTimestamp) = pair_
+                .currentCumulativePrices();
+
+            require(
+                (blockTimestamp != blockTimestampOld),
+                "Oracle: blockTimestamp doesn't change"
+            );
```

Comment:
I think we can remove this check, as there may be situations where we need to get the price in the same transaction as the update. This check will not allow us to do that